### PR TITLE
[DLS] Rename "permissions sync" to "access control sync"

### PR DESCRIPTION
**Pull request by timgrein**
## Related to https://github.com/elastic/enterprise-search-team/issues/4662

This PR changes the name "permissions sync" to "access control sync"

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

**Comment by timgrein** [comment]
buildkite test this

**Review by tarekziade — DISMISSED** [review]

File: connectors/kibana.py [hunk 1]
```diff
@@ -148,11 +148,11 @@ async def prepare(service_type, index_name, config, connector_definition=None):
             "status": "configured",
             "language": "en",
             # Last sync
-            "last_sync_status": None,
             "last_access_control_sync_status": None,
+            "last_access_control_sync_scheduled_at": None,
```
Comment (wangch079) [inline connectors/kibana.py:152]: I think, still not correctly sorted

**Review by wangch079 — APPROVED** [review]

**Review by wangch079 — CHANGES_REQUESTED** [review]
Should also rename `JobType` enum.

**Comment by timgrein** [comment]
> Should also rename `JobType` enum.

Good catch! Adjusted 👍

**Review by wangch079 — APPROVED** [review]

**Pull request merged**